Tests/TuistSupportTests/Utils/GlobTests.swift

```diff
@@ -0,0 +1,317 @@
+        FileManager.default.changeCurrentDirectoryPath(tmpDir.path)
+        defer {
+            FileManager.default.changeCurrentDirectoryPath(originalPath)
+        }
+
+        testWithPrefix("./", pattern: pattern, behavior: behavior, expected: expected)
+    }
+
+    private func testWithPrefix(_ prefix: String, pattern: String, behavior: Glob.Behavior, expected: [String]) {
+        let glob = Glob(pattern: "\(prefix)\(pattern)", behavior: behavior)
+        XCTAssertEqual(
+            glob.paths,
+            expected.map { "\(prefix)\($0)" },
+            "pattern \"\(pattern)\" failed with prefix \"\(prefix)\""
+        )
+    }
+
+    func testBraces() {
+        let pattern = "\(tmpDir.path)/ba{r,y,z}"
+        let glob = Glob(pattern: pattern)
+        var contents = [String]()
+        for file in glob {
+            contents.append(file)
+        }
+        XCTAssertEqual(contents, ["\(tmpDir.path)/bar", "\(tmpDir.path)/baz"], "matching with braces failed")
+    }
+
+    func testNothingMatches() {
+        let pattern = "\(tmpDir.path)/nothing"
+        let glob = Glob(pattern: pattern)
+        var contents = [String]()
+        for file in glob {
+            contents.append(file)
+        }
+        XCTAssertEqual(contents, [], "expected empty list of files")
+    }
+
+    func testDirectAccess() {
+        let pattern = "\(tmpDir.path)/ba{r,y,z}"
+        let glob = Glob(pattern: pattern)
+        XCTAssertEqual(glob.paths, ["\(tmpDir.path)/bar", "\(tmpDir.path)/baz"], "matching with braces failed")
+    }
+
+    func testIterateTwice() {
+        let pattern = "\(tmpDir.path)/ba{r,y,z}"
+        let glob = Glob(pattern: pattern)
+        var contents1 = [String]()
+        var contents2 = [String]()
+        for file in glob {
+            contents1.append(file)
+        }
+        let filesAfterOnce = glob.paths
+        for file in glob {
+            contents2.append(file)
+        }
+        XCTAssertEqual(contents1, contents2, "results for calling for-in twice are the same")
+        XCTAssertEqual(glob.paths, filesAfterOnce, "calling for-in twice doesn't only memoizes once")
+    }
+
+    func testIndexing() {
+        let pattern = "\(tmpDir.path)/ba{r,y,z}"
+        let glob = Glob(pattern: pattern)
+        guard glob.count == 2 else {
+            return XCTFail("Exptected 2 results")
+        }
+        XCTAssertEqual(glob[0], "\(tmpDir.path)/bar", "indexing")
+    }
+
+    // MARK: - Globstar - Bash v3
+
+    func testGlobstarBashV3NoSlash() {
+        // Should be the equivalent of "ls -d -1 /(tmpdir)/**"
+        test(
+            pattern: "**",
+            behavior: GlobBehaviorBashV3,
+            expected: ["bar", "baz", "dir1/", "foo"]
+        )
+    }
+
+    func testGlobstarBashV3WithSlash() {
+        // Should be the equivalent of "ls -d -1 /(tmpdir)/**/"
+        test(
+            pattern: "**/",
+            behavior: GlobBehaviorBashV3,
+            expected: ["dir1/"]
+        )
```

Comment:
```suggestion
        XCTAssertEqual(temporaryDirectory.glob("**"), [temporaryDirectory.appending(component: "dir1"])
```

If we use the Tuist primitives, can't this be rewritten to something like this? I'd try to get rid of the helper `test` method to make the test more straitghtforward